src/parser/Parser.ts

```diff
@@ -1102,6 +1128,36 @@ export class Parser {
             return new Stmt.Return(tokens, toReturn);
         }
 
+        /**
+         * Parses a `goto` statement
+         * @returns an AST representation of an `goto` statement.
+         */
+        function labelStatement() {
+            let tokens = {
+                identifier: advance(),
+                colon: advance(),
+            };
+
+            consume("Labels must be declared on their own line", Lexeme.Newline, Lexeme.Eof);
```

Comment:
Since we're already checking for newlines or EOF following a label, we might be able to drop the `checkNextNext()` function entirely!